package.json

```diff
@@ -45,27 +46,33 @@
     "**/__tests__/fixtures/"
   ],
   "private": true,
+  "lint-staged": {
+    "*.{js,jsx}": [
+      "eslint --fix",
+      "git add"
+    ],
+    "*.{json,md,scss,yaml,yml}": [
+      "prettier --write",
+      "git add"
+    ]
+  },
   "scripts": {
     "bootstrap": "npm-run-all -s check-versions lerna-prepare",
     "check-versions": "babel-node scripts/check-versions.js",
-    "format": "npm-run-all -p format-packages format-cache-dir format-www format-integration format-examples format-scripts format-markdown format-yaml",
-    "format-cache-dir": "prettier-eslint --write \"packages/gatsby/cache-dir/*.js\"",
-    "format-examples": "prettier-eslint --write \"examples/**/gatsby-node.js\" \"examples/**/gatsby-config.js\" \"examples/**/src/**/*.js\"",
-    "format-markdown": "prettier --write \"**/*.md\" --no-semi",
-    "format-packages": "prettier-eslint --write \"packages/*/src/**/*.js\"",
-    "format-scripts": "prettier-eslint --write \"scripts/**/*.js\"",
-    "format-www": "prettier-eslint --write \"www/*.js\" \"www/src/**/*.js\"",
-    "format-integration": "prettier-eslint --write \"integration-tests/**/src/**/*.js\" \"integration-tests/**/cypress/**/*.js\"",
-    "format-yaml": "prettier --write \"**/*.y?(a)ml\"",
+    "format": "npm-run-all -p format-packages format:js format:other",
+    "format:other": "yarn prettier --write",
+    "format:js": "eslint --ignore-path .gitignore \"**/*.{js,jsx}\" --fix",
     "jest": "jest",
     "lerna": "lerna",
     "lerna-prepare": "lerna run prepare",
     "lint": "npm-run-all --continue-on-error --parallel lint:js lint:yaml",
     "lint:js": "eslint --ext .js,.jsx packages/**/src integration-tests/**/src",
     "lint:flow": "babel-node scripts/flow-check.js",
-    "lint:yaml": "prettier --list-different \"**/*.y?(a)ml\"",
+    "lint:other": "yarn prettier --list-different",
     "plop": "plop",
     "prebootstrap": "yarn",
+    "precommit": "lint-staged",
+    "prettier": "prettier \"**/*.{json,md,scss,yaml,yml}\"",
```

Comment:
I more descriptive name for this script would be good I think.

Reply:
any ideas? it's just a prettier command with all the supported files (except js, because it's handled with ESLint/Prettier)

Reply:
`format-code` is nice and descriptive. How about that?

Reply:
But it's an actual `prettier` command.. and it formats everything except the actual code :)

Reply:
lint and format are doing different things, and the difference is the `--write` and `--list-different`, so if we remove that we will have to duplicate the supported files and it's better to have that in as few places as possible.